Automattic/woocommerce-payments
# Refactor to split the attach intent logic with the logic to complete …

Fixes #3407 

#### Changes proposed in this Pull Request


#### A small refactor aimed at separating the responsibilities handled by the function which attaches intent info, with the goal of adding the intent info earlier in the order lifecycle in a follow up PR.



This aims to ensure the `attach_intent_info_to_order` function is only responsible for attaching intent info, and adds another function responsible for adding the order note and marking the payment as complete. 





#### Testing instructions



The testing should ensure everything works in the same way as it did previously.



* Make an order with both a normal product and a subscription product
* Make the orders using both the UPE checkout and the classic checkout experience
* Also test one of the UPE payment methods such as GiroPay.
* Use both the normal checkout and the blocks checkout
* All orders should succeed and have the intent info attached to them, and the correct order note (check the order in WC Admin and verify it is possible to refund the order using WC Payments.

Adding a note here to say that @malinajirka kindly tested the terminals/in-person payments flow in this Slack conversation: p1641819076036300-slack-CGGCLBN58

-------------------


**Post merge**
